File: bench/programs/oo_class.go
package main

import "fmt"

type Point struct{ x, y int }

//go:noinline
func (p *Point) add(o *Point) *Point { p.x += o.x; p.y += o.y; return p }

//go:noinline
func (p *Point) norm() int { return p.x*p.x + p.y*p.y }

func run(n int) int {
	a := &Point{0, 0}
	b := &Point{1, 2}
	s := 0
	for i := 0; i < n; i++ {
		a.add(b)
		s += a.norm()
	}
	return s
}

func main() { fmt.Println(run(400000)) }

File: bench/programs/oo_class.py
class Point:
    def __init__(self, x, y): self.x = x; self.y = y
    def add(self, o): self.x += o.x; self.y += o.y; return self
    def norm(self): return self.x * self.x + self.y * self.y
def run(n):
    a = Point(0, 0); b = Point(1, 2); s = 0
    for i in range(n):
        a.add(b); s += a.norm()
    return s
print(run(400000))
